plugins/dun_gw_bluez4.c

```diff
@@ -90,11 +91,12 @@ static void dun_gw_connect_cb(GIOChannel *io, GError *err, gpointer user_data)
 		return;
 	}
 
-	/* Pick the first powered modem */
-	modem = modems->data;
+	/* Pick the last powered modem */
+	last = g_list_last(modems);
```

Comment:
That can happen for the same reasons as stated in the previous comment (see gprs_watch()).